pkg/file/pdf.go

```diff
@@ -0,0 +1,24 @@
+package file
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+)
+
+// pdfMagic is the [file signature][1] (or magic bytes) for PDF
+//
+// [1]: https://en.wikipedia.org/wiki/List_of_file_signatures
+var pdfMagic = []byte{0x25, 0x50, 0x44, 0x46}
+
+// CheckPDF checks if the provided bytes are a PDF file.
+func CheckPDF(pdf io.Reader) error {
+	buf := make([]byte, len(pdfMagic))
+	if _, err := io.ReadFull(pdf, buf); err != nil {
+		return fmt.Errorf("reading magic bytes: %w", err)
```

Comment:
Nit, but `if errors.Is(err, io.ErrUnexpectedEOF) { return ErrInvalidType }` (that is, it didn't even have 4 bytes, so it can't be a pdf).

Reply:
you mentioned this in the original suggestion, I meant to do it and then got distracted by a merge conflict.

Thinking more about it, I don't even think it's a nit, by making that error be `ErrInvalidType` we will be returning a `4xx` error instead of a `5xx` which is more accurate.

thanks and sorry for missing the original suggestion!